src/execution/mod.rs

```diff
@@ -147,6 +132,10 @@ where
     resume_progress.write(progress_init.clone());
     progress_writer.write(progress_init);
 
+    // Remember! Never build different numbers of Timely operators on
+    // different workers! Timely does not like that and you'll see a
+    // mysterious `failed to correctly cast channel` panic. You must
+    // build asymmetry within each operator.
```

Comment:
Thanks for adding the full error and explanation here, it will be nice next time someone hits the error :)